### PR TITLE
General updates to analytics documentation.

### PR DESCRIPTION
**Pull request by UmanShahzad**
These are pretty simple changes:

1. Broken links
2. Random tiny & simple linting issues.
3. Slight reorganization / additions to the `vars-analytics.yml` file which were necessary for me to make when bringing up an instance.

I have not yet added troubleshooting information related to what issues I ran into. I think because this PR has gotten big I'll make a separate PR for that.

### CODE REVIEW

File: docs/analytics/jenkins.md [hunk 1]
```diff
@@ -210,14 +212,14 @@ Also, ensure these repositories are cloned and readable by the jenkins user:
 ## jenkins_env
 
 Running pipeline jobs requires some environment variables to be set - these are listed in
-[jenkins_env](resources/jenkins_env).  Variables include the [S3 buckets created above](#s3-buckets) and others:
+[jenkins_env](resources/jenkins_env).  Variables include the [S3 buckets created above](#pipeline-s3-buckets) and others:
```
Comment (pomegranited) [inline docs/analytics/jenkins.md:215]: Please change the [referenced section](https://github.com/open-craft/openedx-deployment/pull/7/files#diff-7f455a44f79c304247b8df1cdf8dc667R171)'s title to `Pipeline S3 buckets`.

File: docs/analytics/resources/vars-analytics.yml [hunk 2]
```diff
@@ -131,7 +132,7 @@ JENKINS_ANALYTICS_CREDENTIALS:
 # See AWS_analytics_setup.md#jenkins-seed-job
 ANALYTICS_SCHEDULE_SECURE_REPO_URL: "git@github.com:xxx/secure-configuration.git"
 ANALYTICS_SCHEDULE_SECURE_REPO_VERSION: "master"
-ANALYTICS_SCHEDULE_JOBS_DSL_REPO_URL: "git@github.com:edx-ops/edx-jenkins-job-dsl.git"
+ANALYTICS_SCHEDULE_JOBS_DSL_REPO_URL: "git@github.com:edx/jenkins-job-dsl.git"
```
Comment (pomegranited) [inline docs/analytics/resources/vars-analytics.yml]: This is not the correct repo URL.  Unfortunately, the repo is still private  (cf [jenkins.md#jenkins-seed-job](https://github.com/UmanShahzad/openedx-deployment/blob/cb31b9c97881044a15b5bcdd82be3be079e47206/docs/analytics/jenkins.md#jenkins-seed-job)), and so returns a 404 if you don't have access.

But it has been moved to `git@github.com:edx/jenkins-job-dsl-internal.git, if you could please update it to that.

**Review by pomegranited — APPROVED** [review]
@UmanShahzad Thank you so much for your thorough formatting fixes and link checking!  This documentation was created in a rush, and really needed this level of care.

:+1: once comments are addressed.

- [x] <strike>I tested this</strike>, but I did check the instance that was created using these instructions.
- [x] I read through the code
- [x] <strike>I checked for accessibility issues</strike>
- [x] Includes documentation, and makes it much better.

File: docs/analytics/resources/vars-analytics.yml (hunk 2 above)
Comment (UmanShahzad) [reply docs/analytics/resources/vars-analytics.yml]: Ah! It'll be great when they make this public.

File: docs/analytics/jenkins.md (hunk 1 above)
Comment (UmanShahzad) [reply docs/analytics/jenkins.md:215]: Woops, thanks!

**Comment by UmanShahzad** [comment]
@pomegranited Done!

**Comment by pomegranited** [comment]
Beauty!  Thanks, @UmanShahzad .  Merging now.

**Pull request merged**